### PR TITLE
store/tikv: move lock test to /tests

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
store/tikv has dependencies on `mockstore` and `unistore` because tests need to use them. We can resolve it by moving all tests to standaline package.

Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- move test files into `store/tikv/tests`

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/tests/lock_test.go [hunk 1]
```diff
@@ -673,18 +670,16 @@ func (s *testLockSuite) TestBatchResolveTxnFallenBackFromAsyncCommit(c *C) {
...
-func randKV(keyLen, valLen int) (string, string) {
-	const letters = "abc"
-	k, v := make([]byte, keyLen), make([]byte, valLen)
-	for i := range k {
-		k[i] = letters[rand.Intn(len(letters))]
-	}
-	for i := range v {
-		v[i] = letters[rand.Intn(len(letters))]
+// deduplicateKeys deduplicate the keys, it use sort instead of map to avoid memory allocation.
+func deduplicateKeys(keys [][]byte) [][]byte {
```
Comment (AndreMouche) [inline store/tikv/tests/lock_test.go:674]: The duplicate function makes me unhappy to give LGTM

Comment (disksing) [reply store/tikv/tests/lock_test.go:674]: sorry I'll find a workaround to make you happy again.

Comment (disksing) [reply store/tikv/tests/lock_test.go]: test removed.

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by andylokandy — APPROVED** [review]

**Pull request merged**